spec/models/offer_spec.rb

```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+RSpec.describe Offer, type: :model do
+  describe 'DB table' do
+    it { is_expected.to have_db_column :id }
+    it { is_expected.to have_db_column :name }
+    it { is_expected.to have_db_column :email }
+    it { is_expected.to have_db_column :location }
+    it { is_expected.to have_db_column :price }
+  end
+ 
+  describe 'Validations' do
+    it { is_expected.to validate_presence_of :name }
+    it { is_expected.to validate_presence_of :email }
+    it { is_expected.to validate_presence_of :location }
+    it { is_expected.to validate_presence_of :price }
+  end
+ 
```

Comment:
I need you to test the association between the offer and listing

Reply:
done